src/DynamoCoreWpf/Utilities/ActionDebouncer.cs

```diff
@@ -36,6 +37,23 @@ public void Debounce(int timeout, Action action)
             Cancel();
             cts = new CancellationTokenSource();
 
+            // The TaskScheduler.FromCurrentSynchronizationContext() exists only if there is a valid SyncronizationContex.
+            // Calling this method from a non UI thread could have a null SyncronizationContex.Current,
+            // so in that case we use the default TaskScheduler which uses the thread pool.
+            TaskScheduler taskScheduler = null;
+            if (SynchronizationContext.Current != null)
+            {// This should always be the case in UI threads.
+                taskScheduler = TaskScheduler.FromCurrentSynchronizationContext();
+            }
+            else
+            {// This might happen when running tests in non UI threads.
+                if (Dispatcher.CurrentDispatcher != null)
```

Comment:
shouldn't this check for null?

Reply:
`Dispatcher.CurrentDispatcher != null` - this means it is in the UI thread.
The check is like this:
IF the current sync context not null - then all is as expected.
If the current sync context is null - then we are probably in a unit test (non UI thread)
                                                     - but if we are in a UI thread, then this case is weird and we should log it
Does this make sense?

Reply:
How does `Dispatcher.CurrentDispatcher != null` ensure it's the UI thread?
I thought that this just accessed the dispatcher of the thread?

taken from SO:

```
Dispatcher.CurrentDispatcher gets the dispatcher for the current thread. So, if you're looking for the UI thread's Dispatcher from a background process, don't use this.

Application.Current.Dispatcher will always give you the UI thread's dispatcher, as this is the thread that spins up the sole Application instance.
```

Reply:
Yeah, I wasn't too sure about that. Looks like `Application.Current?.Dispatcher != null` would be a better check for a UI thread,

Reply:
https://stackoverflow.com/questions/5143599/detecting-whether-on-ui-thread-in-wpf-and-winforms